module/PowerNSX.psm1

```diff
@@ -16618,6 +16618,84 @@ function Remove-NsxSslVpnClientInstallationPackage {
     end {}
 }
 
+
+function Remove-NsxSslVpn {
+
+    <#
+    .SYNOPSIS
+    Remove the global SSLVPN configuration of an existing NSX Edge Services
+    Gateway.
+
+    .DESCRIPTION
+    An NSX Edge Service Gateway provides all NSX Edge services such as firewall,
+    NAT, DHCP, VPN, load balancing, and high availability. Each NSX Edge virtual
+    appliance can have a total of ten uplink and internal network interfaces and
+    up to 200 subinterfaces.  Multiple external IP addresses can be configured
+    for load balancer, site‐to‐site VPN, and NAT services.
+
+    SSL VPN allows remote users to connect securely to private networks behind an
+    NSX Edge Services gateway and access servers and applications
+    in the private networks.
+
+    The Remove-NsxSslVpn cmdlet unconfigures the global SSLVPN configuration of
+    the specified Edge Services Gateway.
+
+    .EXAMPLE
+
+    PS C:\> Get-NsxEdge Edge01 | Get-NsxSslVpn | Remove-NsxSslVpn -confirm:$false
+
+    Remove all NSX SSL VPN configuration without confirmation
+
+    #>
+
+    [Diagnostics.CodeAnalysis.SuppressMessageAttribute("PSAvoidDefaultValueSwitchParameter","")] # Cant remove without breaking backward compatibility
+    param (
+
+        [Parameter (Mandatory=$true,ValueFromPipeline=$true,Position=1)]
+            #NSX Edge SslVpn to remove
+            [ValidateScript({ ValidateEdgeSslVpn $_ })]
+            [System.Xml.XmlElement]$SslVpn,
+        [Parameter (Mandatory=$False)]
+            #Prompt for confirmation.  Specify as -confirm:$false to disable confirmation prompt
+            [switch]$Confirm=$true,
```

Comment:
Please add a -NoConfirm switch that behaves opposite to -confirm.  See set-nsxedgefirewall for example.  Since moving to gallery and trying to adhere better to the requirements of PSScriptanalyzer, we need to deprecate the use of -confirm (this is the reason for the PSAvoidDefaultValueSwitchParameter decoration you have included).  Ensure you dont miss the significance of the ParameterSet Name on -confirm and -noconfirm.

People expect -Confirm to be there, so we need to keep adding it to new cmdlets, but I want the default to be -NoConfirm from now on.  Make sure you include the code in begin{} to present the user with a deprecation warning.  In the near future Ill be retrofitting -NoConfirm to existing cmdlets...  Any PR's here are more than welcome! :)